pandas/core/arrays/arrow/array.py

```diff
@@ -1083,6 +1090,24 @@ def pyarrow_meth(data, skip_nulls, **kwargs):
 
         if name in ["min", "max", "sum"] and pa.types.is_duration(pa_type):
             result = result.cast(pa_type)
+        if name in ["median", "mean"] and pa.types.is_temporal(pa_type):
+            result = result.cast(pa_type)
+        if name in ["std", "sem"] and pa.types.is_temporal(pa_type):
+            result = result.cast(pa.int64())
+            if pa.types.is_duration(pa_type):
+                result = result.cast(pa_type)
+            elif pa.types.is_time(pa_type):
+                # TODO: with time types we should probably retain "unit",
+                #  but not clear how to get that since pa_type.unit raises
```

Comment:
Is this on an older version of pyarrow?

```
In [6]: t = pa.time32("ms")

In [7]: t.unit
Out[7]: 'ms'

In [8]: pa.__version__
Out[8]: '10.0.1'

In [9]: t = pa.time64("ns")

In [10]: t.unit
Out[10]: 'ns'

In [11]: t = pa.time32("ms")

In [12]: t.unit
Out[12]: 'ms'
```

Reply:
looks like it depends on exactly what we're checking.  Here pa_type is defined as `pa_type = self._data.type` which  `DataType(time32[ms])`, which does not have a .unit attribute in 10.0.1

Reply:
so how do I get the `Time32Type(time32[ms])` instead of the `DataType(time32[ms])` here?

Reply:
Yeah, this was only recently fixed (should be in pyarrow 11): https://github.com/apache/arrow/pull/14633

When getting the type from the array, it wasn't correctly boxed in the Time specific class, and so the unit attribute is not available. I am not directly sure of a workaround for older pyarrow versions, except for parsing that from the str repr of the type ...

Reply:
so something like?

```
def get_unit_from_pyarrow_dtype(pa_dtype):
    if pa_lt_11:
        return str(pa_dtype).split("[")[-1][:-1]
    return pa_dtype.unit
```

Reply:
Yes, that should work I think